src/plugins/intel_gna/src/backend/gna_limitations.cpp

```diff
@@ -21,6 +21,31 @@ namespace GNAPluginNS {
 namespace GNALimitations {
 namespace Cnn2D {
 
+bool IsEqualToLimit::isValid(const uint32_t val) const {
+    return val == compared_value;
+}
+
+std::string IsEqualToLimit::GetErrorOrEmpty(const uint32_t val) const {
+    std::ostringstream out;
+    if (!isValid(val)) {
+        out << "Unsupported " << what << ", actual value: " << val << ", but should be equal to " << compared_value
+            << "\n";
+    }
+    return out.str();
+}
+
+bool IsLessThanLimit ::isValid(const uint32_t val) const {
+    return val < compared_value;
+}
+
+std::string IsLessThanLimit ::GetErrorOrEmpty(const uint32_t val) const {
```

Comment:
Maybe there is a sense to check range where min is 0?

Reply:
It make sense, and was considering to use `struct RangeLimit`,
but decided to create a separate class (`IsLessThanLimit`) which uses "less operator" (<) instead of "less than or equal" (<=) used by `RangeLimit`.
For most cases `RangeLimit` would be fine. 
However, (in case of kernel size vs padding size limitation) 
would need to use something like:
`RangeLimit padding_size_limit {0, kernel_size - 1};`
for `kernel_size == 0`
`kernel_size - 1` overflows (to MAX_uint32), resulting in `padding_size_limit` accepting any `padding_size`.
Of course `kernel_size == 0` means that something is already wrong, anyway I didn`t want to risk overflow

plus `IsLessThanLimit` is more efficient than `RangeLimit` ;)